### ISSUE TITLE
Image Label Button 

### ISSUE THREAD

**Issue by mey1k** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#1908]
hi, i want to use custom imagebtn(basedon materialbtn)

below is my code
-----------------------------------------------------------------------------------------------------
        <Button Style="{StaticResource MaterialDesignImageButton}"  Template="{StaticResource ImageBtnTemplate}" Content="Hello world" ></Button>
-------------------------------------------------------------------------------------------------------
    <Style x:Key="MaterialDesignImageButton" TargetType="Button" BasedOn="{StaticResource MaterialDesignRaisedAccentButton}">
        <Setter Property="BorderBrush" Value="white" />
        <Setter Property="Background" Value="white"/>
        <Setter Property="Opacity" Value="0.7"/>
        <Style.Triggers>
            <Trigger Property="IsMouseOver" Value="true">
                <Setter Property="BorderBrush" Value="#9F2241"/>
                <Setter Property="Background" Value="#9F2241"/>
            </Trigger>
        </Style.Triggers>
    </Style>
-----------------------------------------------------------------------------------
    <ControlTemplate x:Key="ImageBtnTemplate" TargetType="Button">
            <StackPanel Orientation="Horizontal" Background="{x:Null}">
                <Image Source="{DynamicResource ResourceKey=Img}" Height="50" Width="50"></Image>
                <Label x:Name="lbl" Content="{TemplateBinding Content}" FontWeight="Bold"></Label>
            </StackPanel>
        <ControlTemplate.Triggers>
            <Trigger Property="IsMouseOver" Value="True">
                <Setter Property="Foreground" Value="white" TargetName="lbl"></Setter>
            </Trigger>
        </ControlTemplate.Triggers>
    </ControlTemplate>

----------------------------------------------------------------------------------------------------

When I apply the ControlTemplate, the style disappears.
why ? :(

**Comment by Keboo** [comment]
The reason is because when you apply a control template, it replaces the material design control template. I think what you want to do is to use the `ContentTemplate` property of the `Button` rather than the `Template` property. This will let you define a `DataTemplate` for your content that contains both the `Label` and the `Image`

**Comment by mey1k** [comment]
@Keboo Thank you for reply
I understand what you say.

below is my code
-----------------------------------------------------------------------------------------------------
<Button Style="{StaticResource MaterialDesignBtn}" ContentTemplate="{StaticResource imgButtonTemplate}"/>
-------------------------------------------------------------------------------------------------------
    <Style x:Key="MaterialDesignImageButton" TargetType="Button" BasedOn="{StaticResource MaterialDesignRaisedAccentButton}">
        <Setter Property="BorderBrush" Value="white" />
        <Setter Property="Background" Value="white"/>
        <Setter Property="Opacity" Value="0.7"/>
        <Style.Triggers>
            <Trigger Property="IsMouseOver" Value="true">
                <Setter Property="BorderBrush" Value="#9F2241"/>
                <Setter Property="Background" Value="#9F2241"/>
            </Trigger>
        </Style.Triggers>
    </Style>
-----------------------------------------------------------------------------------
        <DataTemplate x:Key="imgButtonTemplate">
        <StackPanel>
            <TextBlock Text="{Binding Text}"/>
            <Image Source="{Binding Source}"/>
        </StackPanel>
    </DataTemplate>

----------------------------------------------------------------------------------------------------

I don't want to use code behind.

Is there a way to set the content in xaml?

**Comment by w-syss** [comment]
Hi, could you rephrase that question? I am not sure which issue you're having.
What content do you want to set?

**Comment by mey1k** [comment]
@w-syss Hi~:)

I want to implement ImageTextButton.
so, I have to set Image and Text in Button.

but, I can't use ControlTemplate because I use Material Style.

so, I try to implement it using datatemplate.

<DataTemplate x:Key="imgButtonTemplate">
    <StackPanel>
        <TextBlock Text="{Binding Text}"/>
        <Image Source="{Binding Source}"/>
    </StackPanel>
</DataTemplate>

Here, I don't know how to set the Text and Image data in the code above.
Plus, I don't want to use Codebehind(.cs).

Is there a way?

**Comment by bombipappoo** [comment]
Is this what you want?
```
    <Button Style="{StaticResource MaterialDesignImageButton}"
            ContentTemplate="{StaticResource imgButtonTemplate}"
            Content="{Binding}">
```
